src/engine/strat_engine/engine.rs

```diff
@@ -165,40 +178,45 @@ impl Engine for StratEngine {
                     .expect("We just retrieved or created a HashMap");
                 devices.insert(device, dev_node);
 
-                let rc = if let Ok((pool_name, pool)) =
-                    StratPool::setup(&dm, pool_uuid, &devices) {
-                    // We know we have a unique uuid, but the pools table requires a unique name too
-                    // so we will ensure unique before we insert as we don't want to change the
-                    // existing state if we have a conflict.
-                    //
-                    // We will also _not_ exit the daemon as we do in initialize as we were
-                    // previously up and running for some duration of time.
-                    if !self.pools.contains_name(&pool_name) {
-                        self.pools.insert(pool_name, pool_uuid, pool);
-                        Some(pool_uuid)
-                    } else {
-                        let dev_paths = devices
-                            .values()
-                            .map(|p| p.to_str().expect("Expecting valid device path!"))
-                            .collect::<Vec<&str>>()
-                            .join(", ");
-
-                        self.incomplete_pools.insert(pool_uuid, devices);
-
-                        error!("udev add: duplicate pool name {:?} for uuid {:?}, \
+                let rc = match StratPool::setup(&dm, pool_uuid, &devices) {
+                    Ok((pool_name, pool)) => {
+                        // We know we have a unique uuid, but the pools table requires a unique
+                        // name too so we will ensure unique before we insert as we don't want to
+                        // change the existing state if we have a conflict.
+                        //
+                        // We will also _not_ exit the daemon as we do in initialize as we were
+                        // previously up and running for some duration of time.
+                        if !self.pools.contains_name(&pool_name) {
+                            self.pools.insert(pool_name, pool_uuid, pool);
+                            Some(pool_uuid)
+                        } else {
+                            let dev_paths = devices
+                                .values()
+                                .map(|p| p.to_str().expect("Expecting valid device path!"))
```

Comment:
While you're changing all this code around, could you leave out the exclamation mark? We don't use it anywhere else in our ```expect```s. They are supposed to be statements explaining why the developer was confident when they wrote it that the code was logically incapable of failing. If you're not confident, then you'll need to do something more sophisticated to make it possible for the ```let-``` statement to evaluate to an error.